keps/sig-instrumentation/4785-resource-state-metrics/README.md

```diff
@@ -0,0 +1,1216 @@
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+At the moment any disruptions will be indicated by the managed resources' `status` field, like so,
+
+```
+status:                                                                                                     
+    conditions:                                                                  
+    - lastTransitionTime: "2024-08-27T19:46:13Z"
+      message: 'Resource failed to process'
+      observedGeneration: 1                                                                               
+      reason: EventHandlerFailed                                                                                               
+      status: True                                                                                                           
+      type: Failed
+```
+
+Additionally, diagnostic metrics and traces are [planned] for the near-future.
+
+[planned]: https://github.com/rexagod/resource-state-metrics#todo
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+TBD (when targeting beta).
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+The controller aims to completely replace Kube State Metrics' Custom Resource
+State feature-set, and as such, cause it to be deprecated once the KEP graduates
+to stable.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+This is not nota workload feature, but an out-of-tree telemetry solution.
+
+###### How can someone using this feature know that it is working for their instance?
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+- [x] Events: Events are emitted in `EMIT_NAMESPACE` (defaults to ``), for e.g., 
```

Comment:
Events should be emitted in the same namespace as the application that produces them